js/platform-window.js

```diff
@@ -209,8 +217,9 @@ class TitleBar extends HTMLElement {
 
         });
 
-        fin.Platform.getCurrentSync().on('window-context-changed', evt => {
-            if (evt.context) {
+        fin.Platform.getCurrentSync().on('window-context-changed', async (evt) => {
+            const context = await fin.Platform.getCurrentSync().getWindowContext();
+            if (context === void 0 || evt.context && evt.context.theme && evt.context.theme !== context.theme) {
```

Comment:
I think this check is somewhat redundant no? From the setTheme fn:
```js
       ...
       const context = await fin.Platform.getCurrentSync().getWindowContext() || {};
        if (context.theme !== theme) {
            fin.Platform.getCurrentSync().setWindowContext({theme});
        }
```
In any case, perhaps a quick comment to clarify the moving pieces. Something like: "here we listen for all window context events across the entire platform, if we do not have a theme set and any window has set a theme, we use that as our theme as well."
But with better words :)

Reply:
actually this can change significantly, I wrote this in  this way because we had a bug in the core that has since been fixed, I can take another stab at it as part of cleanup.